### PR TITLE
Adding adminpanel

### PR DESCRIPTION
**Pull request by dmahmouali**
https://github.com/threefoldtech/js-sdk/issues/2

### CODE REVIEW

File: jumpscale/packages/admin/actors/admin.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+from jumpscale.servers.gedis.baseactor import BaseActor, actor_method
+from jumpscale.god import j
+from jumpscale.core.exceptions import JSException
+
+explorers = {"main": "explorer.grid.tf", "testnet": "explorer.testnet.grid.tf"}
+
+class Admin(BaseActor):
+
+    @actor_method
+    def admin_list(self) -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/admin.py:10]: list_admin verb comes first

File: jumpscale/packages/admin/actors/admin.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return j.data.serializers.json.dumps({"data": 
+            list({'name':name} for name in j.core.identity.me.admins)})
+               
+    @actor_method
+    def admin_add(self, name: str) -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/admin.py:15]: add_admin

File: jumpscale/packages/admin/actors/admin.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        j.core.identity.me.admins.append(name)
+        return "Added"
+
+    @actor_method
+    def admin_delete(self, name: str) -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/admin.py:20]: delete_admin

File: jumpscale/packages/admin/actors/admin.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+class Admin(BaseActor):
+
+    @actor_method
+    def admin_list(self) -> str:
+        return j.data.serializers.json.dumps({"data": 
+            list({'name':name} for name in j.core.identity.me.admins)})
+               
+    @actor_method
+    def admin_add(self, name: str) -> str:
+        j.core.identity.me.admins.append(name)
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/admin.py:16]: should call .save() ?

File: jumpscale/packages/admin/actors/admin.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        return "Added"
+
+    @actor_method
+    def admin_delete(self, name: str) -> str:
+        j.core.identity.me.admins.remove(name)
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/admin.py:21]: should call .save() ?

File: jumpscale/packages/admin/actors/alerts.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
+from jumpscale.servers.gedis.baseactor import BaseActor, actor_method
+from jumpscale.god import j
+
+class Alerts(BaseActor):
+
+    @actor_method
+    def alerts_list(self) -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/alerts.py:7]: conventions, verb first

File: jumpscale/packages/admin/actors/alerts.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        """
+            get all alerts
+        """
+        ret = [ alert.json for alert in j.tools.alerthandler.find() ]
+        return j.data.serializers.json.dumps({"data":ret})
+
+    @actor_method
+    def alerts_count(self) -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/alerts.py:15]: conventions, verb first

File: jumpscale/packages/admin/actors/alerts.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        """
+            get count of alerts
+        """
+        return j.data.serializers.json.dumps( {"data":j.tools.alerthandler.count()} )
+
+
+    
+    @actor_method
+    def alerts_delete(self, ids: list = [], identifiers: list = []):
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/alerts.py:24]: conventions, verb first

File: jumpscale/packages/admin/actors/alerts.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            delete list of alerts
+        """
+        try:
+            if ids:
+                for _id in ids:
+                    j.tools.alerthandler.delete(_id)
+            elif identifiers:
+                for _identifier in identifiers:
+                    j.tools.alerthandler.delete(_identifier)
+        except Exception as e:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/alerts.py:35]: usage of jumpscale exceptions ?

File: jumpscale/packages/admin/actors/alerts.py [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+            raise e
+
+    @actor_method
+    def alerts_delete_all(self):
+        """
+            delete all alerts
+        """
+        try:
+            j.tools.alerthandler.delete_all()
+        except Exception as e:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/alerts.py:45]: usage of jumpscale exceptions ?

File: jumpscale/packages/admin/actors/alerts.py (hunk 8 above)
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/alerts.py:24]: should be passed as list = None not [] security reasons

File: jumpscale/packages/admin/actors/health.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+        disk_obj = j.sals.fs.shutil.disk_usage('/')
+        res["total"] = disk_obj.total // (1024.0 ** 3)
+        res["used"] = disk_obj.used // (1024.0 ** 3)
+        res["free"] = disk_obj.free // (1024.0 ** 3)
+        res["percent"] = (res["used"] / res["total"]) * 100
+        return j.data.serializers.json.dumps(res)
+
+    @actor_method
+    def health(self) -> str:
+        return "All is good"
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/health.py:18]: what does it do ? if it's just static could be frontend ?

File: jumpscale/packages/admin/actors/packages.py [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+from jumpscale.god import j
+from jumpscale.core.exceptions import JSException
+
+
+class Packages(BaseActor):
+    def __init__(self):
+        self.threebot = j.servers.threebot.get("default")
+
+    @actor_method
+    def packages_get_status(self, names: list) -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/packages.py:11]: conventions too and it returns static string ?

File: jumpscale/packages/admin/actors/packages.py [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+        return "hello from packages_get_status actor"
+
+    @actor_method
+    def packages_list(self) -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/packages.py:15]: conventions too

File: jumpscale/packages/admin/actors/packages.py [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+        return j.data.serializers.json.dumps({"data": self.threebot.packages.get_packages()})
+
+    @actor_method
+    def packages_names(self) -> str:
+        return j.data.serializers.json.dumps({"data": list(self.threebot.packages.list_all())})
+
+    @actor_method
+    def package_add(self, path: str = "", giturl: str = "") -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/packages.py:23]: conventions too

File: jumpscale/packages/admin/actors/packages.py [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+        return j.data.serializers.json.dumps({"data": self.threebot.packages.add(path=path, giturl=giturl)})
+
+    @actor_method
+    def package_delete(self, name: str) -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/packages.py:27]: conventions too

File: jumpscale/packages/admin/actors/solutions.py [hunk 16]
```diff
@@ -0,0 +1,15 @@
+from jumpscale.servers.gedis.baseactor import BaseActor, actor_method
+from jumpscale.god import j
+from jumpscale.core.exceptions import JSException
+
+class Solutions(BaseActor):
+
+    @actor_method
+    def solutions_list(self, name: str = "") -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/solutions.py:8]: conventions too and incomplete ?

File: jumpscale/packages/admin/actors/solutions.py [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+        return "hello from solutions actor"
+               
+    @actor_method
+    def solution_delete(self, name: str) -> str:
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/solutions.py:12]: same as above

File: jumpscale/packages/admin/actors/wallet.py [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+        try:
+            wallet.activate_through_threefold_service()
+        except Exception:
+            j.clients.stellar.delete(name=name)
+            raise j.exceptions.JSException("Error on wallet activation")
+        
+        trustlines = _NETWORK_KNOWN_TRUSTS[str(wallet.network.name)].copy()
+        for asset_code in trustlines.keys():
+            wallet.add_known_trustline(asset_code)
+
```
Comment (waleedhammam) [inline jumpscale/packages/admin/actors/wallet.py:28]: wallet.save() ?

File: jumpscale/packages/admin/frontend/frontend_src/sources/views/main.js [hunk 19]
```diff
@@ -0,0 +1,304 @@
+import { JetView, plugins } from "webix-jet";
+import { admin } from "../services/admin";
+
+export default class TopView extends JetView {
+    config() {
+        const _ = this.app.getService("locale")._;
```
Comment (waleedhammam) [inline jumpscale/packages/admin/frontend/frontend_src/sources/views/main.js:6]: what that men ?

File: jumpscale/packages/admin/package.toml [hunk 20]
```diff
@@ -0,0 +1,8 @@
+name = "admin"
+ports = [80]
```
Comment (waleedhammam) [inline jumpscale/packages/admin/package.toml:2]: add 443 too

File: jumpscale/packages/codeserver/package.py [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+
+    def uninstall(self):
+        """Called when package is deleted
+        """
+        pass
+
+    def start(self):
+        """Called when threebot is started
+        """
+        self.startupcmd.start()
```
Comment (waleedhammam) [inline jumpscale/packages/codeserver/package.py:28]: start only if it's not started ?
maybe make another method called running to check that ?

File: jumpscale/packages/codeserver/package.toml [hunk 22]
```diff
@@ -0,0 +1,12 @@
+name = "codeserver"
+ports = [80]
```
Comment (waleedhammam) [inline jumpscale/packages/codeserver/package.toml:2]: 443 too ?

File: jumpscale/packages/farmmanagement/package.toml [hunk 23]
```diff
@@ -0,0 +1,8 @@
+name = "farmmanagement"
+ports = [ 80,]
```
Comment (waleedhammam) [inline jumpscale/packages/farmmanagement/package.toml:2]: add 443

File: jumpscale/packages/tfgrid_solutions/chats/ubuntu_deploy.py [hunk 24]
```diff
@@ -187,14 +187,9 @@ def container_pay(self):
                 channel_port=self.user_form_data["Logs Channel port"],
                 channel_name=self.user_form_data["Logs Channel name"],
             )
-        metadata = dict()
-        metadata["chatflow"] = self.user_form_data["chatflow"]
-        metadata["Solution name"] = self.user_form_data["Solution name"]
-        metadata["Version"] = self.user_form_data["Version"]
-        metadata["Solution expiration"] = self.user_form_data["Solution expiration"]
```
Comment (waleedhammam) [inline jumpscale/packages/tfgrid_solutions/chats/ubuntu_deploy.py:194]: why this was removed ? duplicate ?

File: jumpscale/packages/admin/actors/admin.py [hunk 25]
```diff
@@ -0,0 +1,72 @@
+from jumpscale.servers.gedis.baseactor import BaseActor, actor_method
+from jumpscale.god import j
+from jumpscale.core.exceptions import JSException
+
+explorers = {"main": "explorer.grid.tf", "testnet": "explorer.testnet.grid.tf"}
```
Comment (ranatrk) [inline jumpscale/packages/admin/actors/admin.py:5]: what about devnet?

File: jumpscale/packages/tfgrid_solutions/package.toml [hunk 26]
```diff
@@ -1,2 +1,2 @@
 name = "tfgrid_solutions"
-ports = [ 80, 443]
+ports = [ 80,]
```
Comment (waleedhammam) [inline jumpscale/packages/tfgrid_solutions/package.toml:2]: please rebase with development_sdk_solutions branch

File: jumpscale/packages/admin/actors/admin.py [hunk 27]
```diff
@@ -0,0 +1,72 @@
...
+        }})
+
+    @actor_method
+    def get_explorer(self) -> str:
+        current_url = j.core.identity.me.explorer_url.strip().lower().split("/")[2]
+        if current_url == explorers["testnet"]:
+            explorer_type = "testnet"
+        elif current_url == explorers["main"]:
+            explorer_type = "main"
+        else:
```
Comment (ranatrk) [inline jumpscale/packages/admin/actors/admin.py:38]: url could also be `devnet`

File: jumpscale/packages/admin/actors/wallet.py [hunk 28]
```diff
@@ -0,0 +1,83 @@
...
+                trustlines.pop(balance.asset_code)
+        for asset_code in trustlines.keys():
+            wallet.add_known_trustline(asset_code)
+
+        return j.data.serializers.json.dumps({"data": trustlines})
+    
+    @actor_method
+    def import_wallet(self, name: str, secret: str, network: str) -> str:
+        network = network or "TEST"
+        wallet = j.clients.stellar.new(name=name, secret=secret, network=network)
```
Comment (ranatrk) [inline jumpscale/packages/admin/actors/wallet.py:75]: should we save the wallet here? or its supposed to be on the go and not saved locally?

File: jumpscale/packages/admin/frontend/frontend_src/sources/views/main.js [hunk 29]
```diff
@@ -0,0 +1,304 @@
...
+                    id: "deployed_gitea",
+                    value: '<span><img class="solutions-icon" src="static/img/gitea.png"/>Gitea</span>'
+                }, {
+                    id: "deployed_domain_delegation",
+                    value: '<span><img class="solutions-icon" src="static/img/domain.png"/>Domain Delegation</span>'
+                }, {
+                    id: "deployed_solution_expose",
+                    value: '<span><img class="solutions-icon" src="static/img/wan.png"/>Solution Expose</span>'
+                }, {
+                    id: "deployed_gateway_4to6",
```
Comment (ranatrk) [inline jumpscale/packages/admin/frontend/frontend_src/sources/views/main.js:97]: the monitoring solution is not added

File: jumpscale/packages/tfgrid_solutions/chats/ubuntu_deploy.py (hunk 24 above)
Comment (dmahmouali) [reply jumpscale/packages/tfgrid_solutions/chats/ubuntu_deploy.py:194]: no need to filter self.user_form_data , just send it yes

File: jumpscale/packages/tfgrid_solutions/package.toml (hunk 26 above)
Comment (dmahmouali) [reply jumpscale/packages/tfgrid_solutions/package.toml:2]: tmam :ok:

File: jumpscale/packages/admin/frontend/frontend_src/sources/views/main.js (hunk 19 above)
Comment (AhmedSa-mir) [reply jumpscale/packages/admin/frontend/frontend_src/sources/views/main.js:6]: Not need now thnx

File: jumpscale/packages/admin/actors/admin.py (hunk 25 above)
Comment (AhmedSa-mir) [reply jumpscale/packages/admin/actors/admin.py:5]: Devnet is done manually not from admin dashboard

File: jumpscale/packages/admin/actors/wallet.py (hunk 28 above)
Comment (AhmedSa-mir) [reply jumpscale/packages/admin/actors/wallet.py:77]: done

File: jumpscale/packages/codeserver/package.py (hunk 21 above)
Comment (AhmedSa-mir) [reply jumpscale/packages/codeserver/package.py]: done

File: jumpscale/packages/admin/frontend/frontend_src/sources/views/main.js (hunk 29 above)
Comment (dmahmouali) [reply jumpscale/packages/admin/frontend/frontend_src/sources/views/main.js:98]: Added :+1:

**Pull request merged**